arm/Microsoft.EventHub/namespaces/deploy.bicep

```diff
@@ -88,24 +80,28 @@ param cuaId string = ''
 @description('Generated. Do not provide a value! This date value is used to generate a SAS token to access the modules.')
 param baseTime string = utcNow('u')
 
+@description('Optional. The event hubs to deploy into this namespace')
+param eventHubs array = []
+
 var maxNameLength = 50
-var uniqueEventHubNamespaceUntrim = '${uniqueString('EventHub Namespace${baseTime}')}'
-var uniqueEventHubNamespace = ((length(uniqueEventHubNamespaceUntrim) > maxNameLength) ? substring(uniqueEventHubNamespaceUntrim, 0, maxNameLength) : uniqueEventHubNamespaceUntrim)
-var constructedNamespaceName = (empty(namespaceName) ? uniqueEventHubNamespace : namespaceName)
-var defaultAuthorizationRuleId = resourceId('Microsoft.EventHub/namespaces/AuthorizationRules', constructedNamespaceName, 'RootManageSharedAccessKey')
+var uniqueEventHubNamespaceUntrim = uniqueString('EventHub Namespace${baseTime}')
+var uniqueEventHubNamespace = length(uniqueEventHubNamespaceUntrim) > maxNameLength ? substring(uniqueEventHubNamespaceUntrim, 0, maxNameLength) : uniqueEventHubNamespaceUntrim
+var name_var = empty(name) ? uniqueEventHubNamespace : name
 var defaultSASKeyName = 'RootManageSharedAccessKey'
-var authRuleResourceId = resourceId('Microsoft.EventHub/namespaces/authorizationRules', constructedNamespaceName, defaultSASKeyName)
-var maximumThroughputUnits_var = ((!isAutoInflateEnabled) ? 0 : maximumThroughputUnits)
+var authRuleResourceId = resourceId('Microsoft.EventHub/namespaces/authorizationRules', name_var, defaultSASKeyName)
+var maximumThroughputUnits_var = !isAutoInflateEnabled ? 0 : maximumThroughputUnits
 var virtualNetworkRules = [for index in range(0, (empty(networkAcls) ? 0 : length(networkAcls.virtualNetworkRules))): {
   id: '${vNetId}/subnets/${networkAcls.virtualNetworkRules[index].subnet}'
 }]
 var networkAcls_var = {
-  bypass: (empty(networkAcls) ? json('null') : networkAcls.bypass)
-  defaultAction: (empty(networkAcls) ? json('null') : networkAcls.defaultAction)
-  virtualNetworkRules: (empty(networkAcls) ? json('null') : virtualNetworkRules)
-  ipRules: (empty(networkAcls) ? json('null') : ((length(networkAcls.ipRules) == 0) ? json('null') : networkAcls.ipRules))
+  bypass: empty(networkAcls) ? null : networkAcls.bypass
+  defaultAction: empty(networkAcls) ? null : networkAcls.defaultAction
+  virtualNetworkRules: empty(networkAcls) ? null : virtualNetworkRules
+  ipRules: empty(networkAcls) ? null : ((length(networkAcls.ipRules) == 0) ? null : networkAcls.ipRules)
 }
-var namespaceAlias_var = (empty(namespaceAlias) ? 'placeholder' : namespaceAlias)
+
+@description('Optional. The disaster recovery config for this namespace')
+param disasterRecoveryConfig object = {}
```

Comment:
Should we try to keep the bicep file structured to have:
1. params
1. vars
1. resources/modules
1. outputs

In other words, move params over all the var declarations

Reply:
yes we should

Reply:
done